lib/brexit_checker/results_audiences.rb

```diff
@@ -0,0 +1,56 @@
+class BrexitChecker::ResultsAudiences
+  class << self
+    def populate_business_groups(audience_actions, selected_criteria)
+      return [] if audience_actions.blank? || selected_criteria.blank?
+
+      {
+        actions: audience_actions,
+        criteria: filter_criteria_by_actions(audience_actions, selected_criteria),
+      }
+    end
+
+    def populate_citizen_groups(audience_actions, selected_criteria)
+      return [] if audience_actions.blank? || selected_criteria.blank?
+
+      all_possible_groups = BrexitChecker::Groups.load_all
+      grouped_actions = all_possible_groups.each_with_index.map do |group, group_index|
+        actions_in_group = filter_actions_by_group(audience_actions, group.key)
```

Comment:
you could move the ```filter_actions_by_group``` to the Group model, so you could use ```group.actions``` What do you think?

Reply:
humm, not sure what that would look like.
Could have `group.filter_actions_by_group(audience_actions)` and essentially move this from ResultsAudiences to groups. Possibly renaming it.

But the group model is pretty stupid, it just knows about it's YAML file and nothing else. We need to take a preselected array of audience_actions here and group them by examining those actions. Does it make sense to move it from the class that's all about formatting the results page data structures?